### PR TITLE
🚀 feat(metric): added histogram of anomaly scores

### PR DESCRIPTION
**Pull request by abc-125**
## 📝 Description

I have updated the existing `AnomalyScoreDistribution` to optionally collect labels and use them to plot a histogram of image-level scores of the distribution using these labels.

Example of histogram generation with data imitating the real distribution of scores:

```python
from anomalib.metrics import AnomalyScoreDistribution
import torch

preds_good = torch.distributions.normal.Normal(loc=0.3, scale=0.1).sample([1000]) 
target_good = torch.zeros_like(preds_good)

preds_bad = torch.distributions.normal.Normal(loc=0.7, scale=0.1).sample([100]) 
target_bad = torch.ones_like(preds_bad)

dist = AnomalyScoreDistribution()
dist.update(anomaly_scores=preds_good, labels=target_good)
dist.update(anomaly_scores=preds_bad, labels=target_bad)
dist.generate_figure()
```

hist

## ✨ Changes

- Added optional collection of labels, a function to generate a figure, and updated examples in `src/anomalib/metrics/anomaly_score_distribution.py`
- Added function to plot histogram of scores to `src/anomalib/metrics/utils.py`

Select what type of change your PR is:

- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔄 Refactor (non-breaking change which refactors the code base)
- [ ] ⚡ Performance improvements
- [ ] 🎨 Style changes (code style/formatting)
- [ ] 🧪 Tests (adding/modifying tests)
- [ ] 📚 Documentation update
- [ ] 📦 Build system changes
- [ ] 🚧 CI/CD configuration
- [ ] 🔧 Chore (general maintenance)
- [ ] 🔒 Security update
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).
- [x] 🏷️ My PR title follows conventional commit format.

For more information about code review checklists, see the [Code Review Checklist](https://github.com/open-edge-platform/anomalib/blob/main/docs/source/markdown/guides/developer/contributing.md).

### CODE REVIEW

File: src/anomalib/metrics/anomaly_score_distribution.py [hunk 1]
```diff
@@ -116,3 +128,43 @@ def compute(self) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tenso
             self.pixel_std = anomaly_maps.std(dim=0).squeeze()
 
         return self.image_mean, self.image_std, self.pixel_mean, self.pixel_std
+
+    def generate_figure(
```
Comment (samet-akcay) [inline src/anomalib/metrics/anomaly_score_distribution.py]: Would it be an idea to rename this to `plot`. this would be inline with the naming convention of the rest of the methods as well as torchmetrics 
https://lightning.ai/docs/torchmetrics/stable/pages/plotting.html

Comment (samet-akcay) [inline src/anomalib/metrics/anomaly_score_distribution.py]:
```suggestion
    def plot(
```

Comment (abc-125) [reply src/anomalib/metrics/anomaly_score_distribution.py]: Sure, thank you for the suggestion! I have changed the name.

File: src/anomalib/metrics/anomaly_score_distribution.py [hunk 2]
```diff
@@ -116,3 +128,43 @@ def compute(self) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tenso
...
+        bins: int = 30,
+        good_color: str = "skyblue",
+        bad_color: str = "salmon",
+        xlabel: str = "Score",
+        ylabel: str = "Relative Count",
+        title: str = "Score Histogram",
+        legend_labels: tuple[str, str] = ("Good", "Bad"),
+    ) -> tuple[Figure, str]:
+        """Generate a histogram of scores.
+
```
Comment (rajeshgangireddy) [inline src/anomalib/metrics/anomaly_score_distribution.py:143]: Although it's easy to follow, for convention, could you expand the doc string for this public method: 

```
Args:
    bins (int, optional): Number of histogram bins. Defaults to 30.
    good_color (str, optional): Color for 'good' samples. Defaults to "skyblue".
    bad_color (str, optional): Color for 'bad' samples. Defaults to "salmon".
    xlabel (str, optional): Label for the x-axis. Defaults to "Score".
    ylabel (str, optional): Label for the y-axis. Defaults to "Relative Count".
    title (str, optional): Title of the plot. Defaults to "Score Histogram".
    legend_labels (tuple[str, str], optional): Legend labels for good and bad samples. Defaults to ("Good", "Bad").
```

Comment (abc-125) [reply src/anomalib/metrics/anomaly_score_distribution.py:143]: Sure, thank you for noticing this, I usually try to follow conventions 😅

**Review by rajeshgangireddy — DISMISSED** [review]
Thanks for the feature.

**Review by samet-akcay — APPROVED** [review]
thanks for adding this nice feature
